src/diffpy/utils/parsers/loadmetafile.py

```diff
@@ -0,0 +1,212 @@
+#                   (c) 2010 The Trustees of Columbia University
+#                   in the City of New York.  All rights reserved.
+#
+# File coded by:
+#
+# See AUTHORS.txt for a list of people who contributed.
+# See LICENSE_DANSE.txt for license information.
+#
+##############################################################################
+
+import pathlib
+import json
+
+# FIXME: add support for yaml, xml
+supported_formats = ['.json']
+
+
+def load_PDF_into_db(dbname, pdfname, hddata: dict, rv: list, show_path=True):
+    """Load an entry consisting of PDF header and base data into a database file.
+
+    Requires hdata and rv generated from loadData.
+
+    dbname          -- name of the database file to load an entry into.
+    pdfname         -- name of the PDF file.
+    hddata          -- Dictionary of PDF metadata generated by loadData.
+    rv              -- List of PDF (r, gr) pairs generated by loadData.
+    show_path       -- include a PDF_path element in the database entry (default True).
+
+    Returns the dictionary loaded from/into the updated database file.
+    """
+    # new file or update
+    existing = False
+    if pathlib.Path.is_file(dbname):
+        existing = True
+
+    # collect entry
+    with open(pdfname, 'r') as grfile:
+        data = {}
+
+        # add path
+        grpath = grfile.name
+        if show_path:
+            data.update({'PDF_path': grpath})
+
+        # add r, gr, and header metadata
+        data.update({'r': list(rv[:, 0]), 'gr': list(rv[:, 1])})
+        data.update(hddata)
+
+        # parse name using pathlib and generate json entry
+        name = pathlib.Path(grpath).name
+        entry = {name: data}
+
+    # check if supported type
+    extension = pathlib.Path(dbname).suffix
+    if extension not in supported_formats:
+        raise Exception(f"Format of {dbname} is not supported.")
+
+    # json
+    if extension == '.json':
+        # dump if non-existing
+        if not existing:
+            with open(dbname, 'w') as jsonfile:
+                pdfs = entry  # for return
+                json.dump(pdfs, jsonfile, indent=2)
+
+        # update if existing
+        else:
+            with open(dbname, 'r') as json_read:
+                pdfs = json.load(json_read)
+                pdfs.update(entry)
+            with open(dbname, 'w') as json_write:
+                # dump to string first for formatting
+                json.dump(pdfs, json_write, indent=2)
+
+    return pdfs
+
+
+def markup_PDF(hddata: dict, rv: list, muname=None):
+    # FIXME: may be better suited for REST API package, not diffpy.utils
+    """Put PDF file information into a dictionary.
+
+    hddata  -- Dictionary of metadata.
+    rv      -- List of (r, gr) pairs.
+    muname  -- file to save into (default None, no saving occurs).
+
+    Returns the dictionary loaded from/into markup file.
+    """
+
+    # gather data
+    data = {}
+    data.update({'r': list(rv[:, 0]), 'gr': list(rv[:, 1])})
```

Comment:
since we are making it more abstract than just pdf files, maybe change to x and y instead of r and g?

Also, what happens if loadData reads more columns than just 2.  Do we lose the other columns?

Reply:
Yes, changed. See dt_colnames parameter in docstring.